### ISSUE TITLE
Inconsistent AddonType for "io" bundles in Karaf & Marketplace add-on services

### ISSUE THREAD

**Issue by ghys** [openhab/openhab-core#2507]
The "io" bindings are set to the "io" AddonType in the CommuntiyMarketplaceAddonService:

https://github.com/openhab/openhab-core/blob/1d17166c79c79dfff5cbb4e96803a7d7d67ccf8f/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java#L103

https://github.com/openhab/openhab-core/blob/1d17166c79c79dfff5cbb4e96803a7d7d67ccf8f/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityBundleAddonHandler.java#L43-L44

While in the Karaf add-on service they are set as "misc":

https://github.com/openhab/openhab-core/blob/1d17166c79c79dfff5cbb4e96803a7d7d67ccf8f/bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafAddonService.java#L59

@wborn that's why https://community.openhab.org/t/log4j2-extra/127214 didn't appear in the "Other" tab of the UI when you added the "io" tag (it displays the "misc" add-on under "System Integrations", not "io").

**Comment by wborn** [comment]
Yes that's why I switched the category and was also looking into what happend. :+1:
